### PR TITLE
add & customize divise gem

### PR DESCRIPTION
**Pull request by nguyenluc312**

### CODE REVIEW

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -9,7 +9,11 @@
 <body>
   <%= render "layouts/header" %>
   <div class="container">
+    <% flash.each do |message_type, message| %>
+      <div class="alert alert-<%= message_type %>"><%= message %></div>
+    <% end %>
     <%= yield %>
+  <%= debug params if Rails.env.development? %>
```
Comment (dogiadat) [inline app/views/layouts/application.html.erb]: indent

**Comment by dogiadat** [comment]
OK

**Comment by hieuhtk58** [comment]
ok

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -2,4 +2,17 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+  before_action :configure_permitted_parameters, if: :devise_controller?
+
+  def configure_permitted_parameters
```
Comment (dieunb) [inline app/controllers/application_controller.rb]: private function

**Pull request merged**